evalml/tests/component_tests/test_featuretools.py

```diff
@@ -163,3 +164,55 @@ def test_dfs_sets_max_depth_1(mock_dfs, X_y_multi):
     feature.fit(X_pd, y)
     _, kwargs = mock_dfs.call_args
     assert kwargs["max_depth"] == 1
+
+
+@patch("evalml.pipelines.components.transformers.preprocessing.featuretools.dfs")
+def test_dfs_with_serialized_features(mock_dfs, X_y_binary):
+    X, y = X_y_binary
+    X_pd = pd.DataFrame(X)
+    X_pd.columns = X_pd.columns.astype(str)
+    X_fit = X_pd.iloc[: len(X) // 3]
+    X_transform = X_pd.iloc[len(X) // 3 :]
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X", dataframe=X_transform, index="index", make_index=True
+    )
+    feature_matrix, features = ft.dfs(
+        entityset=es, target_dataframe_name="X", trans_primitives=["absolute"]
+    )
+
+    feature = DFSTransformer(features=features)
+    feature.fit(X_fit)  # no-op
+    assert not mock_dfs.called
+
+    X_t = feature.transform(X_transform)
+    assert_frame_equal(feature_matrix, X_t)
+    assert features == feature.features
+
+
+@patch("evalml.pipelines.components.transformers.preprocessing.featuretools.dfs")
+@patch(
+    "evalml.pipelines.components.transformers.preprocessing.featuretools.calculate_feature_matrix"
+)
+def test_dfs_skip_transform(mock_calculate_feature_matrix, mock_dfs, X_y_binary):
```

Comment:
Give serialized features and input that contains the transformed columns, this test checks that both DFS and calculate feature matrix is not called.

Reply:
I think this is a typo. Both tests use:

```python
    feature_matrix, features = ft.dfs(
        entityset=es, target_dataframe_name="X", trans_primitives=["absolute"]
    )
    features = list(filter(lambda f: not isinstance(f, IdentityFeature), features))
```

And both create these features:

```python
['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',
       '13', '14', '15', '16', '17', '18', '19', 'ABSOLUTE(0)', 'ABSOLUTE(1)',
       'ABSOLUTE(10)', 'ABSOLUTE(11)', 'ABSOLUTE(12)', 'ABSOLUTE(13)',
       'ABSOLUTE(14)', 'ABSOLUTE(15)', 'ABSOLUTE(16)', 'ABSOLUTE(17)',
       'ABSOLUTE(18)', 'ABSOLUTE(19)', 'ABSOLUTE(2)', 'ABSOLUTE(3)',
       'ABSOLUTE(4)', 'ABSOLUTE(5)', 'ABSOLUTE(6)', 'ABSOLUTE(7)',
       'ABSOLUTE(8)', 'ABSOLUTE(9)']
```